appveyor.yml

```diff
@@ -45,7 +45,7 @@ environment:
    - PROFILE: MSVC2015
      PLATFORM: x64
      CONFIGURATION: Release
-     BOOST_VERSION: 1.64.0
```

Comment:
Why?

Reply:
VS2015 image does not contain Boost 1.64 : it only contains versions up to 1.63.
1.64 and 1.65.1 are included only in VS2017 image. See https://www.appveyor.com/docs/build-environment/#boost for full details

This is not an issue introduced by this bug, but my patch highlights the problem: currently, this build expect Boost 1.64, which is not present, thus the C++ lib is not built.

With my patches, the C++ lib can be built (even without Boost), but the tests still require Boost: so building in this case would require an extra flag to disable tests.

Since the goal is (I think...) to actually build with a recent version of Boost, I downgraded the version the most recent installed version in the image.